woof-code/rootfs-skeleton/etc/profile

```diff
@@ -2,13 +2,41 @@
 #100622 PKG_CONFIG_PATH is determined in 3builddistro (Woof), written to /etc/profile.d/pkgconfig
 #110804 fix double-login when exit from X, allow /etc/profile to complete. see also "echo -n '# '" in /usr/bin/xwin at exit.
 
-PATH="/bin:/usr/bin:/sbin:/usr/sbin:/usr/local/bin:/root/my-applications/bin:/usr/games"
-if [ -f /lib64/libc.so.6 ] ; then #slackware64
-	LD_LIBRARY_PATH="/lib64:/usr/lib64:/root/my-applications/lib:/usr/local/lib"
-else
-	LD_LIBRARY_PATH="/lib:/usr/lib:/root/my-applications/lib:/usr/local/lib"
+[ "$PATH" == "" ] && export PATH="/bin:/sbin:/usr/bin:/usr/sbin:/usr/local/bin:/usr/local/sbin:/usr/X11R7/bin"
+
+if [ "$LD_LIBRARY_PATH" == "" ]; then
+
+	LD_LIBRARY_PATH32="/lib:/lib32:/libx32:/lib/i386-linux-gnu:/usr/lib:/usr/lib32:/usr/libx32:/usr/lib/i386-linux-gnu:/usr/local/lib:/usr/local/lib32:/usr/local/libx32:/usr/local/lib/i386-linux-gnu:/root/my-applications/lib"
+	LD_LIBRARY_PATH64="/lib:/lib64:/lib/x86_64-linux-gnu:/usr/lib:/usr/lib64:/usr/lib/x86_64-linux-gnu:/usr/local/lib:/usr/local/lib64:/usr/local/lib/x86_64-linux-gnu:/root/my-applications/lib"
+	LD_LIBRARY_PATH64_COMPAT="/lib32:/libx32:/lib/i386-linux-gnu:/usr/lib32:/usr/libx32:/usr/lib/i386-linux-gnu:/usr/local/lib32:/usr/local/libx32:/usr/local/lib/i386-linux-gnu"
+	
+	if [ -f /lib64/libc.so.6 ] || [ -f /lib/x86_64-linux-gnu/libc.so.6 ]; then #slackware64
+		LD_LIBRARY_PATH="${LD_LIBRARY_PATH64}:${LD_LIBRARY_PATH64_COMPAT}"
+	elif [ -f /lib/libc.so.6 ]; then
+	  if [ -L /lib/libc.so.6 ]; then
+	    ORIGFILE="$(readlink /lib/libc.so.6 2>/dev/null)"
+	    [ "$(echo "$ORIGFILE" | grep '/')" == "" ] && ORIGFILE="/lib/${ORIGFILE}"
+	    if [ "$(file "$ORIGFILE" | grep "64-bit")" != "" ]; then
+	     LD_LIBRARY_PATH="${LD_LIBRARY_PATH64}:${LD_LIBRARY_PATH64_COMPAT}"
+	    else
+	     LD_LIBRARY_PATH="$LD_LIBRARY_PATH32"
+	    fi
+	  else
+	   	if [ "$(file /lib/libc.so.6 | grep "64-bit")" != "" ]; then
```

Comment:
This will slow down every new shell (= every new terminal, every new terminal tab)